src/serializer.rs

```diff
@@ -133,6 +134,34 @@ where
         };
         Serializer { out, ty, registry }
     }
+
+    fn serialize_compact(&mut self, ty: u32, v: u128) -> Result<()> {
+        let type_def = self
+            .registry
+            .expect("registry not present")
```

Comment:
`expect`(and `unwrap`) should be avoided when possible specially when we are already returning a `Result`.

Reply:
Ended up using `self.resolve`, as it did the same thing as this snippet.

Paradoxically, this function also uses `expect` 🤭.